presto-main/src/main/java/com/facebook/presto/operator/scalar/MergingDigest.java

```diff
@@ -767,155 +660,136 @@ public double quantile(double q) {
                     }
                 };
             }
 
             @Override
-            public int size() {
-                return lastUsedCell;
+            public int size()
+            {
+                return activeCentroids;
             }
         };
     }
 
     @Override
-    public double compression() {
+    public double compression()
+    {
         return publicCompression;
     }
 
     @Override
-    public int byteSize() {
+    public int byteSize()
+    {
         compress();
-        // format code, compression(float), buffer-size(int), temp-size(int), #centroids-1(int),
+        // format code, type code, min (double), max (double), compression(double), #centroids(int),
         // then two doubles per centroid
-        return lastUsedCell * 16 + 32;
-    }
-
-    @Override
-    public int smallByteSize() {
-        compress();
-        // format code(int), compression(float), buffer-size(short), temp-size(short), #centroids-1(short),
-        // then two floats per centroid
-        return lastUsedCell * 8 + 30;
+        return activeCentroids * 16 + 30;
     }
 
     @SuppressWarnings("WeakerAccess")
-    public ScaleFunction getScaleFunction() {
+    public ScaleFunction getScaleFunction()
+    {
         return scale;
     }
 
-    public enum Encoding {
-        VERBOSE_ENCODING(1), SMALL_ENCODING(2);
+    public enum Encoding
+    {
+        VERBOSE_ENCODING((byte) 0), SMALL_ENCODING((byte) 1);
 
-        private final int code;
+        private final byte code;
 
-        Encoding(int code) {
+        Encoding(byte code)
+        {
             this.code = code;
         }
     }
 
     @Override
-    public void asBytes(ByteBuffer buf) {
-        compress();
-        buf.putInt(Encoding.VERBOSE_ENCODING.code);
-        buf.putDouble(min);
-        buf.putDouble(max);
-        buf.putDouble(publicCompression);
-        buf.putInt(lastUsedCell);
-        for (int i = 0; i < lastUsedCell; i++) {
-            buf.putDouble(weight[i]);
-            buf.putDouble(mean[i]);
-        }
-    }
+    public Slice serialize()
+    {
+        SliceOutput sliceBuilder = new DynamicSliceOutput(byteSize());
 
-    @Override
-    public void asSmallBytes(ByteBuffer buf) {
         compress();
-        buf.putInt(Encoding.SMALL_ENCODING.code);    // 4
-        buf.putDouble(min);                          // + 8
-        buf.putDouble(max);                          // + 8
-        buf.putFloat((float) publicCompression);           // + 4
-        buf.putShort((short) mean.length);           // + 2
-        buf.putShort((short) tempMean.length);       // + 2
-        buf.putShort((short) lastUsedCell);          // + 2 = 30
-        for (int i = 0; i < lastUsedCell; i++) {
-            buf.putFloat((float) weight[i]);
-            buf.putFloat((float) mean[i]);
-        }
+        sliceBuilder.writeByte(0); // version 0 of T-Digest serialization                     // 1
```

Comment:
I prefer using `Encoding` as the code used to do. With this change `Encoding` doesn't seem to be used anymore?